src/dataExplorer/context/fields.tsx

```diff
@@ -68,6 +77,37 @@ export const FieldsProvider: FC<Prop> = ({children, scope}) => {
 
     setLoading(RemoteDataState.Loading)
 
+    if (isFlagEnabled('v2privateQueryUI') && isIOx) {
+      // user input is sanitized to avoid SQL injection
+      const sanitized = sanitizeSQLSearchTerm(searchTerm)
+      const queryTextSQL: string = `
+      SELECT column_name
+      FROM information_schema.columns
+      WHERE
+           table_schema = 'iox'
+           AND data_type NOT LIKE 'Dictionary%'
+           AND table_name = '${measurement}'
+           AND column_name != 'time'
+           AND column_name ILIKE '%${sanitized}%'
+      LIMIT ${DEFAULT_LIMIT}
+      `
+      try {
+        const resp = await queryAPI(queryTextSQL, scope, {
+          language: LanguageType.SQL,
+          bucket,
+        } as QueryOptions)
+        const values = (resp.parsed.table?.columns?.column_name?.data ??
+          []) as string[]
```

Comment:
Is the reason we don't get strong typing here (and have to type cast using `as`) that we can't predict the type of `resp.parsed.table.columns.column_name.data`?

Reply:
@wdoconnell We actually can predict the type of `resp.parsed.table.columns.column_name.data` which is `string[] | number[] | boolean[]`. Since field keys are strings, so we cast the data to `string[]` specifically